src/types.h

```diff
@@ -1,14 +1,16 @@
 #include<stdint.h>
 
-typedef struct double_ans_t {
+typedef struct double_ans_t { // used in froll
   double *ans;
-  uint8_t status;   // 0:ok, 1:message, 2:warning, 3:error; unix return signal: {0,1,2}=0, {3}=1
-  char message[4][256]; // STDOUT: output, STDERR: message, warning, error
+  uint8_t status;             // 0:ok, 1:message, 2:warning, 3:error; unix return signal: {0,1,2}=0, {3}=1
+  char message[4][4096];      // STDOUT: output, STDERR: message, warning, error
 } double_ans_t;
+// implicit n_message limit discussed here: https://github.com/Rdatatable/data.table/issues/3423#issuecomment-487722586
 
-typedef struct ans_t {
+typedef struct ans_t {        // used in nafill
   int32_t *int_v;
   double *dbl_v;
-  uint8_t status;   // 0:ok, 1:message, 2:warning, 3:error; unix return signal: {0,1,2}=0, {3}=1
-  char message[4][256]; // STDOUT: output, STDERR: message, warning, error
+  uint8_t status;             // 0:ok, 1:message, 2:warning, 3:error; unix return signal: {0,1,2}=0, {3}=1
+  char message[4][256];       // STDOUT: output, STDERR: message, warning, error
```

Comment:
Should this 256 in `ans_t` be 4096 too, consistent with `double_ans_t`?  The discussion in #3423 is based on `message` allocation of 4096.  The `snprintf` limit is 500 and could feasibly overwrite 256.

Reply:
`ans_t` is currently used only in `nafill.c`, and only single messages are being written there, thus 256 is fine there: https://github.com/Rdatatable/data.table/blob/master/src/nafill.c#L61 (todo `sprintf->snprintf`)
but
we can/**should** drop `double_ans_t` and use `ans_t` instead if **we agree** on the overhead of creating extra integer pointer - `ans_t` creates both double and int pointers. I now think it is very safe, `ans_t` refers to column-rolling_window combination, so unlikely to scale more than few hundreds.

I can follow up in this PR. Change `double_ans_t` to `ans_t`.